### PR TITLE
YA66R (yet another #66 rebase)

### PR DESCRIPTION
**Pull request by quinnj**
@cjprybol, I wasn't able to push to your fork, so it created a new branch. This includes a proper 0.2.0 DataStreams implementation.

### CODE REVIEW

**Comment by nalimilan** [comment]
Let's backup the original branch (if you're afraid of breaking something), and force-push to #66? Better not split the discussion in multiple parts.

File: src/abstractdatatable/io.jl [hunk 1]
```diff
@@ -195,73 +195,90 @@ end
...
+    columns::T
+    header::Vector{String}
 end
+DataTableStream(dt::DataTable) = DataTableStream(Tuple(dt.columns), map(String, names(dt)))
 
-# DataTable as a Data.Source
+# DataTable Data.Source implementation
+function Data.schema(dt::DataTable)
+    return Data.Schema(Type[eltype(A) for A in dt.columns],
+                        map(string, names(dt)), length(dt) == 0 ? 0 : length(dt.columns[1]))
```
Comment (nalimilan) [inline src/abstractdatatable/io.jl:209]: Incorrect indentation. BTW, you can use `string.(names(dt))` here and above.

File: src/abstractdatatable/io.jl [hunk 2]
```diff
@@ -195,73 +195,90 @@ end
...
 
-function DataTable{T <: Data.StreamType}(sch::Data.Schema,
-                                         ::Type{T}=Data.Field,
-                                         append::Bool=false)
-    rows, cols = size(sch)
-    rows = max(0, T <: Data.Column ? 0 : rows) # don't pre-allocate for Column streaming
-    columns = Vector{Any}(cols)
+# Construct or modify a DataTable to be ready to stream data from a source with a schema of `sch`
+function DataTable(sch::Data.Schema{R}, ::Type{S}=Data.Field,
+                    append::Bool=false, args...; reference::Vector{UInt8}=UInt8[]) where {R, S <: Data.StreamType}
```
Comment (nalimilan) [inline src/abstractdatatable/io.jl:233]: Missing space here too.

**Review by nalimilan — COMMENTED** [review]
Could you try to keep lines under 92 chars? That's the style guide we used up to now.

**Comment by cjprybol** [comment]
Thanks, Jacob! I think the issue with pushing to my branch was that I had pushed it to my own fork rather than the main repo. I'll stop doing that to avoid the same issue in the future

**Pull request closed without merging**